src/pages/train/index.tsx

```diff
@@ -23,65 +39,42 @@ const Train: React.FC = () => {
 
   const [subPage, setSubPage] = useState<subPage>("training");
 
-  // exercisePlanId from the url
-  const { exercisePlanId } = useParams();
+  const { data, isLoading } = useGetExerciseByIdQuery(rawExercise.id);
 
   useEffect(() => {
     let isMounted = true;
 
-    if (!loading) return;
-
-    api.execute(Routes.getDoneExercises()).then((response) => {
-      if (!isMounted) return;
-      if (!response.success) {
-        setError(true);
-      }
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      const exerciseData = response.data.exercises.find((e: any) => {
-        return e.exercise_plan_id === parseInt(exercisePlanId ?? "");
+    if (data && isMounted) {
+      setExercise({
+        title: data.title,
+        description: data.description,
+        sets: rawExercise.sets,
+        repeatsPerSet: rawExercise.repeats_per_set,
+        videoPath:
+          data.video ??
+          "https://vid.pr0gramm.com/2021/12/28/130aaef3ab9c207a.mp4",
+        activated: true,
       });
-      if (exerciseData)
-        api
-          .execute(Routes.getExercise({ id: exerciseData.id }))
-          .then((response) => {
-            setExercise({
-              title: response.data.title,
-              description: response.data.description,
-              sets: exerciseData.sets,
-              repeatsPerSet: exerciseData.repeats_per_set,
-              videoPath:
-                response.data.video ??
-                "https://vid.pr0gramm.com/2021/12/28/130aaef3ab9c207a.mp4",
-              activated: response.data.title,
-            } as ExerciseData);
-            setLoading(false);
-          });
-      else {
-        setError(true);
-        setLoading(false);
-      }
-    });
+    }
 
     return () => {
-      // clean up
       isMounted = false;
     };
-  }, [loading, exercisePlanId]);
+  }, [data, rawExercise.repeats_per_set, rawExercise.sets]);
 
   return (
     <Container confimLeave={subPage !== "exerciseDone"}>
-      {loading || exercise === undefined ? (
+      {isLoading || exercise === null ? (
```

Comment:
Wouldn't that also work?
```suggestion
      {isLoading || !exercise ? (
```